stubs/PyYAML/yaml/representer.pyi

```diff
@@ -1,14 +1,16 @@
 import datetime
 import sys
 from types import BuiltinFunctionType, FunctionType, ModuleType
-from typing import Any, Callable, ClassVar, Mapping, NoReturn, Sequence, Tuple, Type, TypeVar
+from typing import AbstractSet, Any, Callable, ClassVar, ItemsView, Mapping, NoReturn, Sequence, Tuple, Type, TypeVar, Union
 
 from yaml.error import YAMLError as YAMLError
 from yaml.nodes import MappingNode as MappingNode, Node as Node, ScalarNode as ScalarNode, SequenceNode as SequenceNode
 
 _T = TypeVar("_T")
 _R = TypeVar("_R", bound=BaseRepresenter)
 
+MappingLike = Union[Mapping[Any, Any], ItemsView[Any, Any], AbstractSet[Tuple[Any, Any]]]
```

Comment:
Looking at the source code, the following is even looser and should cover most cases:

```suggestion
MappingLike = Union[Mapping[Any, Any], Iterable[Tuple[Any, Any]]]
```

(Where `Iterable` needs to be imported from `typing`.) The only thing not covered are custom classes that implement only `items()`, a negligible case.

Reply:
@srittau you're right - I think this is also the case for `represent_{sequence,list,set}`. My concern was that this allows duplicate pairs - if dupes passed, they will be lost on read, but again, this would happen with the impl as well.